leshan-client-demo/src/main/java/org/eclipse/leshan/client/demo/LeshanClientDemo.java

```diff
@@ -608,7 +617,7 @@ public static void createAndStartClient(String endpoint, String localAddress, in
                 initializer.setInstancesForObject(SERVER, new Server(123, lifetime));
             } else if (clientCertificate != null) {
                 initializer.setInstancesForObject(SECURITY, x509(serverURI, 123, clientCertificate.getEncoded(),
-                        clientPrivateKey.getEncoded(), serverCertificate.getEncoded()));
+                        clientPrivateKey.getEncoded(), serverCertificate.getEncoded(), certificateUsage.code));
```

Comment:
Should we add this for Bootstrap case too ? 
Some line above : 
```java
 else if (clientCertificate != null) {
                initializer.setInstancesForObject(SECURITY, x509Bootstrap(serverURI, clientCertificate.getEncoded(),
                        clientPrivateKey.getEncoded(), serverCertificate.getEncoded()));
                initializer.setClassForObject(SERVER, Server.class);
            } 
```

Reply:
Sure -- was wondering why it wasn't there--- actually it was ;) but not in this code set...

Reply:
Updated the PR with it.